### ISSUE TITLE
What about make `prime_server` a submodule?

### ISSUE THREAD

**Issue by playgithub** [valhalla/valhalla#4206]
In my practice, `prime_server` is not in repo of archlinux, if it is a submodule, it can be installed automatically.
BTW `prime_server` has only 50+ stars on github, it is not very hot.

**Comment by kevinkreiser** [comment]
i have also been thinking we should use it as a submodule. there was a pr a long time ago that seemed to have the same idea but its long since defunct. anyway the annoyance with it as a submodule is that we want to use it as more than a library, we also want to build its binaries (`prime_httpd` and `prime_proxyd`). im sure its doable (just need to mess around with cmake a bit.


> BTW prime_server has only 50+ stars on github, it is not very popular.
jeez tough crowd!

**Comment by playgithub** [comment]
> we also want to build its binaries (`prime_httpd` and `prime_proxyd`)

Does `valhalla` need the binaries?

**Comment by xxhh1112** [comment]
t

**Comment by playgithub** [comment]
`prime_server` is not in the official repo of debian, also of archlinux.
Making `prime_server` a submodule of `valhalla` would be easier to install `valhalla`.

**Comment by playgithub** [comment]
`prime_server` is not in the official repo of debian, also of archlinux.
Making `prime_server` a submodule of `valhalla` would be easier to install `valhalla`.

**Comment by kevinkreiser** [comment]
yes valhalla needs the binaries if you want to run the service in an operationally sound way.

im not opposed to doing a submodule, i just havent had time to spend on it

**Comment by kevinkreiser** [comment]
i guess we could submodule+cmake for the regular builds but then when we do the docker build we'd actually build prime_server binaries as well so we can ship them with the rest of valhallas binaries. that seems like it should work unless im missing something :smile:

**Comment by playgithub** [comment]
It's better.

What about integrate all the things of `prime_server` (like `prime_httpd` and `prime_proxyd`) into `valhalla` as libs instead of binaries? If it doesn't make sense feel free to ignore it.
